src/main/java/ch/njol/skript/expressions/ExprStringColor.java

```diff
@@ -0,0 +1,207 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.util.Color;
+import ch.njol.skript.util.ColorRGB;
+import ch.njol.skript.util.SkriptColor;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Name("String Colors")
+@Description({
+	"Retrieve the first, the last, or all of the color objects or color codes of a string.",
+	"The retrieved color codes of the string will be formatted with the color symbol."
+})
+@Examples({
+	"set {_colors::*} to the string colors of \"<red>hey<blue>yo\"",
+	"",
+	"set {_color} to the first string color code of \"&aGoodbye!\"",
+	"send \"%{_color}%Howdy!\" to all players"
+})
+@Since("INSERT VERSION")
+public class ExprStringColor extends PropertyExpression<String, Object> {
+
+	private enum StringColor {ALL, FIRST, LAST;}
```

Comment:
```suggestion
	private enum StringColor {
		ALL, FIRST, LAST
	}
```
should be at the bottom

Reply:
Is this a coding convention?

Reply:
yes

Reply:
Well, then this general java convention must be brand spanking new considering I've seen classes that have it at the top without me ever touching it.

Reply:
We don't have any conventions relating to where inner classes go. Bottom is typical, but sometimes enums are put near the top since they're used in registering/init/etc
Doesn't really matter either way imo.

Reply:
Thank you